tests/components/evohome/test_init.py

```diff
@@ -23,8 +23,9 @@ async def test_entities(
     """Test entities and state after setup of a Honeywell TCC-compatible system."""
 
     # some extended state attrs are relative the current time
-    freezer.move_to("2024-07-10 12:00:00+00:00")
+    freezer.move_to("2024-07-10T12:00:00Z")
 
-    await setup_evohome(hass, config, install=install)
+    async for _ in setup_evohome(hass, config, install=install):
+        await hass.async_block_till_done()
```

Comment:
Why do we need to wait n times instead of only one at the end?

Reply:
This is a long story...

In `async_setup()` there is one or two `hass.async_create_task(async_load_platform(...` and the `async_block_till_done()` in `setup_evohome()` definitely blocks until all the `async_setup_platform()` calls are complete.

However, when `async_setup_platform()` invokes the `async_add_entities` callback, it includes `update_before_add=True`, which causes `ep._async_add_and_update_entities()` to be invoked, instead of `ep._async_add_entities()`.

The former creates further tasks, one for each entity, and then:
```
async with self.hass.timeout.async_timeout(timeout, self.domain):
    results = await asyncio.gather(*tasks, return_exceptions=True)
```

Unfortunately, the first `async_block_till_done()` appears to return *before* this second tranche of tasks are DONE.

Reply:
However, in `test_entities` of **test_init.py**, you notice another.

In `async_setup()` there is one or two `hass.async_create_task(async_load_platform(...` and the `async_block_till_done()` in `setup_evohome()` definitely blocks until all the corresponding `async_setup_platform()` calls are complete.

However, when `async_setup_platform()` invokes the `async_add_entities` callback, it includes `update_before_add=True`, which causes `ep._async_add_and_update_entities()` to be invoked, instead of `ep._async_add_entities()`.

The former creates further tasks, one for each entity, and then:
```
async with self.hass.timeout.async_timeout(timeout, self.domain):
    results = await asyncio.gather(*tasks, return_exceptions=True)
```

Unfortunately, the first `async_block_till_done()` appears to return *before* this second tranche of tasks are DONE.